### ISSUE TITLE
how is multi gpu loss gathered?

### ISSUE THREAD

**Issue by SnapixAI** [kohya-ss/sd-scripts#1558]
I've been looking into the sd3 train branch, im trying to understand how are the loss gathered for multi-gpu and would love to understand the logic behind it.
I'm used to working with accelerator.gather/reduce for loss/tensor updates. however im not seeing any of that being used in the sd3 training script which got me curious - how are the losses gathered across all processes

**Comment by BootsofLagrangian** [comment]
Accelerate does this thing. In [accelerator.accumulate](https://github.com/huggingface/accelerate/blob/159c0dd02a42c30545821b7287376fe4be04d5ee/src/accelerate/accelerator.py#L1046) context manager, accelerate synchronize gradients and loss via [sync_gradients](https://github.com/huggingface/accelerate/blob/159c0dd02a42c30545821b7287376fe4be04d5ee/src/accelerate/accelerator.py#L1020).

sd-scripts utilizes accelerate from Hugging Face, it is very helpful to do high-level distributed learning.
